app/policies/transcription_policy.rb

```diff
@@ -0,0 +1,30 @@
+class TranscriptionPolicy < ApplicationPolicy
+  delegate :editor?, :approver?, :viewer?, to: :project_policy
+
+  def update?
+    admin? || (logged_in? && editor?)
+  end
+
+  def project_policy
+    ProjectPolicy.new(user, projects)
+  end
+
+  def projects
+    # Project.where(id: records.collect(&:workflow).pluck(:project_id).uniq)
+    records.collect(&:workflow).collect(&:project)
```

Comment:
is this an 2xN lookup? 

Would it be better to try something like
`Project.where(workflow_id: records.collect(&:workflow_id))`

Reply:
The commented query above was me trying to determine exactly that, as you can see. I was having quite a time trying to get the SQL to print during testing, and it wasn't clear to me whether the commented query was actually better. 

The `collects` didn't seem to result in any extra queries, leading me to believe that the relation was being eager loaded somewhere further up. I couldn't confirm that, though, so if you feel better about the commented `Project.where` version I can switch.

Reply:
It won't pre / eager load by default, perahps your testing setup has already loaded the associations for those transcription records and thus you don't see the double lookup. 

Also you can use `.distinct` vs ruby land `uniq` to only return the uniq set of projects from the db.

Reply:
Yeah, it was confusing and hard to tell what was gonna be happening for real. I also thought about adding an `include(:whatever)` to the individual controller calls but I'm not sure it would help: Strangely, once you're in the policy, `records` is already cast as array of objects, not an AR relation. For that reason also you gotta use uniq over distinct.

Reply:
So `ActiveRecord::Relation.collect` will return an array class, and even if you pass in a AR:Relation from your query scope, you still wrap that in an array via the `ApplicationPolicy#initialize` method https://github.com/zooniverse/tove/blob/27ab26714d64c96dc4e62c36812c09497e042cc8/app/policies/application_policy.rb#L7

So those records in your policies will always be an array vs something enumerable like AR:Relation

I think this would help with the uniq situation vs discinct via sql.

Reply:
Forgot about that array wrapping. Gonna leave it for now, but I'll inline #projects and update it to do a Project.where.